object_gomap_test.go

```diff
@@ -326,3 +328,154 @@ func TestGoMapUnicode(t *testing.T) {
+			expectedNewMem: SizeEmptyStruct + ((5+SizeString)+SizeInt)*4,
+			errExpected:    nil,
+		},
+		{
+			name:      "should account for each key value pair given map with a nil value",
+			threshold: 100,
+			val: &objectGoMapSimple{
+				baseObject: baseObject{
+					val: &Object{runtime: vm},
+				},
+				data: map[string]interface{}{
+					"test": nil,
+				},
+			},
+			// overhead + len("test") + null
+			expectedMem: SizeEmptyStruct + 4 + SizeEmptyStruct,
+			// overhead + len("test") with string overhead + null
+			expectedNewMem: SizeEmptyStruct + (4 + SizeString) + SizeEmptyStruct,
+			errExpected:    nil,
+		},
+		{
+			name:      "should account for nested reflect object",
+			threshold: 100,
+			val: &objectGoMapSimple{
+				baseObject: baseObject{
+					val: &Object{runtime: vm},
+				},
+				data: map[string]interface{}{
+					"test": &objectGoMapSimple{
+						baseObject: baseObject{
+							val: &Object{runtime: vm},
+						},
+						data: map[string]interface{}{
+							"subTest1": valueInt(99),
+							"subTest2": valueInt(99),
+							"subTest3": valueInt(99),
+							"subTest4": valueInt(99),
+						},
+					},
+				},
+			},
+			// overhead + len("test") + reflectObject
+			expectedMem: SizeEmptyStruct + 4 + SizeEmptyStruct,
+			// overhead + len("test") with string overhead + reflectObject
+			expectedNewMem: SizeEmptyStruct + 4 + SizeString + SizeEmptyStruct,
+			errExpected:    nil,
+		},
+		{
+			name:      "should account for nested key value pairs",
+			threshold: 100,
+			val: &objectGoMapSimple{
+				baseObject: baseObject{
+					val: &Object{runtime: vm},
+				},
+				data: map[string]interface{}{
+					"test": nestedMap,
+				},
+			},
+			// overhead + len("test") + VM runtime base object + values
+			expectedMem: SizeEmptyStruct + 4 + nestedMapMemUsage,
+			// overhead + len("testN") with string overhead + VM runtime base object with overhead + values with string overhead
+			expectedNewMem: SizeEmptyStruct + (4 + SizeString) + nestedMapNewMemUsage,
+			errExpected:    nil,
+		},
+		{
+			name:      "should account for key value pair with value over threshold",
+			threshold: 20,
+			val: &objectGoMapSimple{
+				baseObject: baseObject{
+					val: &Object{runtime: vm},
+				},
+				data: map[string]interface{}{
+					"test": []interface{}{
+						valueInt(99),
+						valueInt(99),
+						valueInt(99),
+						valueInt(99),
+					},
+				},
+			},
+			// overhead + len("testN") + (value + len("length") + "length".value + prototype + ints)
+			expectedMem: SizeEmptyStruct + 4 + (SizeEmptyStruct + 6 + SizeEmptyStruct + (SizeEmptyStruct + SizeEmptyStruct) + SizeNumber*4),
+			// overhead + len("testN") with string overhead + (value + len("length") with string overhead + "length".value + prototype + ints)
+			expectedNewMem: SizeEmptyStruct + (4 + SizeString) + (SizeEmptyStruct + (6 + SizeString) + SizeEmptyStruct + (SizeEmptyStruct + SizeEmptyStruct) + SizeNumber*4),
+			errExpected:    nil,
+		},
```

Comment:
[minor] This test is actually not covering anything around the threshold. The only way to make this have meaning would be to add an estimation logic on the `objectGoMapSimple` `Memusage` function similarly to what we do with `baseObject`.

For context the threshold for objects is to protect us from really large objects where we have to waste CPU running through each k/v pairs. So we sample it and use that to estimate what the usage is.

In this specific test case you would only hit it if the `baseObject` that we embed was above that threshold but that case is already covered in a different test. Alternatively I believe the slice of interfaces gets converted to `objectGoSlice` which is another place where we could use the same array threshold we use for `arrayObject`. But I would expect that to be covered in the `objectGoSlice` tests.

I would remove the threshold test and file a ticket to add estimation for the two types you worked on in this PR.